cpp/dolfin/refinement/ParallelRefinement.cpp

```diff
@@ -29,6 +58,8 @@ ParallelRefinement::ParallelRefinement(const mesh::Mesh& mesh)
   if (!_mesh.topology().connectivity(1, 0))
     throw std::runtime_error("Edges must be initialised");
 
+  _neighbour_comm = create_neighbor_comm_via_edge(mesh);
```

Comment:
I can't see where this communicator get destroyed.

Reply:
Good point. I tend to forget they are not C++ objects with destructors.

Reply:
Memory management of the comms can be a bit tricky. It might be possible to attached a counter each time we taker hold of one, or we can make more use of the comm wrapper.